packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1044,14 +1044,7 @@ export function getExecutionContext(): ExecutionContext {
 }
 
 export function flushDiscreteUpdates() {
-  // TODO: Should be able to flush inside batchedUpdates, but not inside `act`.
-  // However, `act` uses `batchedUpdates`, so there's no way to distinguish
-  // those two cases. Need to fix this before exposing flushDiscreteUpdates
-  // as a public API.
```

Comment:
I don't agree that we shouldn't flush inside `act`. `act` was supposed to test closer to what a user sees. Batching discrete updates is not something a user can ever see.